go/arrow/csv/reader.go

```diff
@@ -76,6 +86,49 @@ func NewReader(r io.Reader, schema *arrow.Schema, opts ...Option) *Reader {
 	return rr
 }
 
+func (r *Reader) readHeader() error {
+	records, err := r.r.Read()
+	if err != nil {
+		return errors.New("failed to read file")
+	}
+
+	var fields []arrow.Field
+	for _, name := range records {
```

Comment:
I don't think that's the correct handling of the header.
AFAIK, CSV header lines just hint at the name of the column of data.
they don't say anything about the type of the column of data.

_e.g._ I believe one could very well have to deal with headers like so:
```
city;population
New-York;8,622,698
Chicago;2,000,001
```

so... all we can do is modify/update the name of the column.
what do you think?

Reply:
oh, I see, sorry for the misunderstanding! Previously I thought we want to encode/decode field type as header. Sounds good.